README.md

```diff
@@ -1,8 +1,49 @@
 # Operations
 
-Welcome to your new gem! In this directory, you'll find the files you need to be able to package up your Ruby library into a gem. Put your Ruby code in the file `lib/operations`. To experiment with that code, run `bin/console` for an interactive prompt.
+## A bit of theory
 
-TODO: Delete this and the text above, and describe your gem
+### What is an operation
+
+First of all, let's define an application as a combination of domain logic and application state. Domain logic can either read and return the parts of the application state to the consumer (Query) or can modify the application state (Command).
+
+**Note:** There is a concept that is called Command Query Separation (SQS) or Command Query Responsibility Segregation (CQRS) and which can be used at any level of the implementation (classes in OOP, API) but the general idea is simply not to mix these two up.
+
+While Query is a simple concept (just fetch data from the application state and render it to the consumer in the requested form), Command implementation can have a lot of caveats.
+
+_Command_ or _business operation_ or _interaction_ or _use case_ or _application service_ (DDD term) even just _service_ (this term is so ambiguous) is a predefined sequence of programmed actions that can be called by a user (directly in the code or via the API) and modifies the application state. In the scope of this framework, we prefer the _operation_ term though.
+
+These modifications are atomic in the sense that the application state is supposed to be consistent and valid after the operation execution. It might be eventually consistent but the idea is that the application state is valid after the operation execution and in a normal case, there should be no need to call a complimentary operation to make the state valid.
+
+Operations can also create different side effects such as: sending an email message, making asynchronous API calls (shoot-and-forget), pushing events to an event bus, etc.
+
+**Note:** An important note is that contrary to the pure DDD approach considers aggregate as a transactional boundary but in this framework - the operation itself is wrapped inside of a transaction, though it is configurable.
```

Comment:
Aggregate was not called for in any previous part of the documentation, it is a bit confusing to find it pop in here without mentions of what an aggregate is/does in relation to the operation framework.

Reply:
this is a DDD aggregate

Reply:
paraphrased it a bit